src/Microsoft.VisualStudio.Threading/DispatcherExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿/********************************************************
+*                                                        *
+*   © Copyright (C) Microsoft. All rights reserved.      *
+*                                                        *
+*********************************************************/
+
+#if NET45
```

Comment:
❓ Does your net45 and netstandard build have the same assembly identity? If so, this condition is setting users up for failure unless the netstandard build has light-up support for the same signature.

Reply:
> so I can't imagine a scenario where folks get runtime failures with this model unless they are mis-applying net45 targeted DLLs onto .NET Core.

We can't use System.Collections.Immutable with Newtonsoft.Json inside Visual Studio 2017 because some unrelated applications GAC'd the .NET 4.0 build of that library and the required light-up layer was not included. If the .NET Standard build of vs-threading has a strong name and valid signature, it's safe to assume that for some users it will be loaded in .NET Framework contexts outside the control of the application.

Reply:
Well then, one idea might be to strong name, but only "OSS sign" the netstandard builds so that they can be used for compilation but never GAC'd. If .NET Framework would allow GAC'ing of a .NET Core build, we could do the same there. 
That would reduce the problem to just net45 vs. net46, for which vs-threading already has light-ups (at least in most cases -- I'm not sure if they're comprehensive).
But then, I've never heard of this *actually* being a problem for this assembly.